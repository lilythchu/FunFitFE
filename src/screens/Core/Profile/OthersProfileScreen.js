import React, {useState} from 'react';
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  ActivityIndicator,
} from 'react-native';
import {Avatar} from '@rneui/themed';
import CustomButton from '../../../components/CustomButton';
import Chevron from '../../../components/Chevron';
import globalStyles from '../../../../styles/global';
import globalColors from '../../../../styles/colors';
import {useNavigation, useRoute} from '@react-navigation/native';
import {arrayToString} from '../../../../utils/methods';
import {avaGender} from '../../../../utils/methods';
import client from '../../../../api/client';

const OthersProfileScreen = () => {
  const {info, token} = useRoute().params;
  const [loading, setLoading] = useState(false);
  const navigation = useNavigation();
  const initConvo = () => {
    setLoading(true);
    client
      .post(
        '/chat/initiateConvo',
        {anotherUserId: info._id},
        {headers: {Authorization: `Bearer ${token}`}},
      )
      .then(res => {
        setLoading(false);
        navigation.navigate('ChatStory');
      })
      .catch(err => {
        navigation.navigate('ChatStory');
      });
  };

  return (
    <ScrollView
      showsHorizontalScrollIndicator={false}
      style={globalStyles.scrollView}>
      <Chevron navigation={navigation} color={globalColors.blueFaded} />

      <View style={{alignSelf: 'center'}}>
        <Avatar
          size={150}
          rounded
          source={avaGender(info.sex)}
          containerStyle={{backgroundColor: 'lightgray'}}
        />
        <View style={styles.nameContainer}>
          <Text style={styles.username}>{info.name}</Text>
          <Text style={styles.bio}>{info.lifestyleTarget}</Text>
        </View>
      </View>

      <View style={styles.infoContainer}>
        <View style={[styles.subInfoContainer, {flex: 1, marginRight: 5}]}>
          <Text style={styles.subTitle}>Level</Text>
          <Text style={styles.text}>{info.level}</Text>
        </View>
        <View style={[styles.subInfoContainer, {flex: 1, marginLeft: 5}]}>
          <Text style={styles.subTitle}>Points</Text>
          <Text style={styles.text}>{info.points}</Text>
        </View>
        <View style={[styles.subInfoContainer, {flex: 1, marginLeft: 5}]}>
          <Text style={styles.subTitle}>Age</Text>
          <Text style={styles.text}>{info.age}</Text>
        </View>
      </View>
      
      <View style={styles.infoContainer}>
        <View style={[styles.subInfoContainer, {flex: 1, marginRight: 5}]}>
          <Text style={styles.subTitle}>Workout Interests</Text>
          <Text style={styles.text}>
            {arrayToString(info.workoutInterests)}
          </Text>
      </View>

      </View>

      <CustomButton type="THIRD" title="Message" onPress={initConvo} loading={loading} />
    </ScrollView>
  );
};

export default OthersProfileScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  username: {
    fontWeight: '500',
    fontSize: 28,
  },
  image: {
    flex: 1,
    width: undefined,
    height: undefined,
  },
  titleBar: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 24,
    marginHorizontal: 16,
  },
  profileImage: {
    width: 150,
    height: 150,
    borderRadius: 100,
    overflow: 'hidden',
  },
  nameContainer: {
    alignSelf: 'center',
    alignItems: 'center',
    marginVertical: 16,
  },
  bio: {
    fontSize: 18,
  },
  infoContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingBottom: 5,
  },
  subInfoContainer: {
    alignItems: 'center',
    // backgroundColor: globalColors.cream,
    borderRadius: 10,
    margin: 10,
    padding: 5,
  },
  subTitle: {
    fontSize: 18,
    fontWeight: '400',
    color: globalColors.username,
  },
  text: {
    fontSize: 14,
    padding: 5,
    textAlign: 'center',
  },
  btnTitle: {
    borderWidth: 1,
    textAlign: 'center',
    padding: 10,
    fontSize: 20,
    borderRadius: 5,
    width: 150,
    alignSelf: 'center',
    borderColor: globalColors.blueFaded,
    color: globalColors.navyBlue,
  },
});
