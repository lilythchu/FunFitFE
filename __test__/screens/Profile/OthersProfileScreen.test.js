import React from 'react';
import {render, screen} from '@testing-library/react-native';
import OthersProfileScreen from '../../../src/screens/Core/Profile/OthersProfileScreen';

const mockedNavigate = jest.fn();
const mockInfo = {
  name: 'friendName',
  workoutInterests: ['yoga'],
  age: '20',
  points: 0,
  level: 1,
}

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useRoute: () => ({
      params: {
        info: mockInfo,
        token: 'token'
      }
    }),
  };
});

describe('Other Profile screen', () => {
  test('renders correctly', () => {
    render(<OthersProfileScreen />);
    expect(screen).toMatchSnapshot();
  })
})