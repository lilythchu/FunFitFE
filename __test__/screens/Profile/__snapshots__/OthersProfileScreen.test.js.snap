// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other Profile screen renders correctly 1`] = `
<RCTScrollView
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "backgroundColor": "white",
      "padding": 20,
    }
  }
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": 20,
          "marginTop": 40,
          "opacity": 1,
        }
      }
      testID="chevron"
    >
      <Text />
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "lightgray",
            "borderRadius": 75,
            "height": 150,
            "width": 150,
          }
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 75,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            borderRadius={75}
            onLoad={[Function]}
            source={1}
            style={
              Object {
                "bottom": 0,
                "flex": 1,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              }
            }
            testID="RNE__Image"
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
            transition={true}
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "center",
                  "width": undefined,
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "marginVertical": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 28,
              "fontWeight": "500",
            }
          }
        >
          friendName
        </Text>
        <Text
          style={
            Object {
              "fontSize": 18,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 5,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 10,
              "margin": 10,
              "padding": 5,
            },
            Object {
              "flex": 1,
              "marginRight": 5,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#458452",
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Level
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          1
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 10,
              "margin": 10,
              "padding": 5,
            },
            Object {
              "flex": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#458452",
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Points
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          0
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 10,
              "margin": 10,
              "padding": 5,
            },
            Object {
              "flex": 1,
              "marginLeft": 5,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#458452",
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Age
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          20
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 5,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 10,
              "margin": 10,
              "padding": 5,
            },
            Object {
              "flex": 1,
              "marginRight": 5,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#458452",
              "fontSize": 18,
              "fontWeight": "400",
            }
          }
        >
          Workout Interests
        </Text>
        <Text
          style={
            Object {
              "fontSize": 14,
              "padding": 5,
              "textAlign": "center",
            }
          }
        >
          yoga
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          Object {
            "marginTop": 10,
          },
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "transparent",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "width": 150,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 20,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Message
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
